ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/util/DataStructureUtil.java

```diff
@@ -538,10 +539,14 @@ public ExecutionPayloadHeader randomExecutionPayloadHeader(final SpecVersion spe
                     .baseFeePerGas(randomUInt256())
                     .blockHash(randomBytes32())
                     .transactionsRoot(randomBytes32())
-                    .withdrawalsRoot(this::randomBytes32)
+                    .withdrawalsRoot(() -> withdrawalsRoot)
```

Comment:
oh nice - does builder listen to the normal `getPayloadV2` calls? Just wondering how they know what withdrawals to include... sorry for the totally naive question, just looking for what gaps we might have...

Reply:
Yes, we fork version the `getPayload` calls. I think builders also run nodes and know the current state and which withdrawals to include. There is a certain bit of trust that once you agree on the bid and send the signed blinded block, they will return a valid execution payload. In case of invalid payload, blocks will be rejected as bad blocks are rejected now, so builders won't get any MEV, so they are not incentivized to send you bad things. Even if they do keep sending bad blocks, at some point, circuit breaker will kick in and disable the builder flow.